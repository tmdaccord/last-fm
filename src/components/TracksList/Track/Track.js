import React from 'react';
import classes from './Track.module.scss';
import trackImg from '../../../assets/images/track.png';
import {Link} from "react-router-dom";

const Track = (props) => {
    const artistLink = props.artistUrl ? <a className={classes.Link} href={props.artistUrl}>last.fm</a> : null;
    return (
        <div className={classes.Track}>
            <div className={classes.Img}>
                <img src={props.imageUrl ? props.imageUrl : trackImg} alt={props.name}/>
            </div>
            <div>
                <h5>{props.name}</h5>
                <h6><Link to={`/artist/${props.artistName}`}>{props.artistName}</Link> {artistLink}</h6>
            </div>
        </div>
    );
};

export default Track;
