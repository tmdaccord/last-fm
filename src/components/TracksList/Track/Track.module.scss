@import "../../../assets/styles/variables";

.Track {
    display: flex;
    align-items: flex-start;
    padding: 15px;
    background-color: lighten(saturate(adjust_hue($contrast, 0), 0%), 25%);
    color: $default;

    > div {
        flex-grow: 1;
    }

    h5 {
        color: $default;
        margin-bottom: 5px;
    }
}

.Link {
    display: inline-block;
    padding: 1px 3px;
    font-size: 10px;
    line-height: 1;
    border-radius: 2px;
    text-transform: none;
    background-color: lighten(saturate(adjust_hue($contrast, 0), 0%), 60%);
    color: $primary;
    vertical-align: baseline;
    &:hover {
        background-color: lighten(saturate(adjust_hue($contrast, 0), 0%), 80%);
    }
}

.Img {
    margin-right: 15px;
    flex-basis: 30%;
    max-width: 30%;
    flex-shrink: 0;

    img {
        width: 100%;
        height: auto;
    }
}
