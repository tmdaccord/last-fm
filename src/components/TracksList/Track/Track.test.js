import React from "react";
import {BrowserRouter as Router} from 'react-router-dom';

import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Track from "./Track";

configure({adapter: new Adapter()});

describe('<Track />', () => {
    let component;
    const props = {
        name: 'name',
        imageUrl: 'imageUrl',
        artistName: 'artistName',
        artistUrl: 'artistUrl'
    };

    beforeEach(() => {
        component = mount(<Router><Track {...props} /></Router>)
    });

    it('render correctly Track component', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render artist link if available', function () {
        expect(component.find('a')).toHaveLength(2);
    });

    it('should not render artist link if artistUrl not available', function () {
        const component = mount(<Router><Track name='name' imageUrl='imageUrl' artistName='artistName' /></Router>);
        expect(component.find('a')).toHaveLength(1);
    });
});
