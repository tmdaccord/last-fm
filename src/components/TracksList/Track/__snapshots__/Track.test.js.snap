// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> render correctly Track component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Track
      artistName="artistName"
      artistUrl="artistUrl"
      imageUrl="imageUrl"
      name="name"
    >
      <div
        className="Track"
      >
        <div
          className="Img"
        >
          <img
            alt="name"
            src="imageUrl"
          />
        </div>
        <div>
          <h5>
            name
          </h5>
          <h6>
            <Link
              to="/artist/artistName"
            >
              <LinkAnchor
                href="/artist/artistName"
                navigate={[Function]}
              >
                <a
                  href="/artist/artistName"
                  onClick={[Function]}
                >
                  artistName
                </a>
              </LinkAnchor>
            </Link>
            <a
              className="Link"
              href="artistUrl"
            >
              last.fm
            </a>
          </h6>
        </div>
      </div>
    </Track>
  </Router>
</BrowserRouter>
`;
